plugin.video.hardstyletv/plugintools.py

```diff
@@ -0,0 +1,582 @@
+            socket.setdefaulttimeout(deftimeout)
+        except:
+            import sys
+            for line in sys.exc_info():
+                _log( "%s" % line )
+    
+    # Actualiza el almacén de cookies
+    cj.save(ficherocookies)
+
+    # Lee los datos y cierra
+    if handle.info().get('Content-Encoding') == 'gzip':
+        buf = StringIO( handle.read())
+        f = gzip.GzipFile(fileobj=buf)
+        data = f.read()
+    else:
+        data=handle.read()
+
+    info = handle.info()
+    _log("read_body_and_headers Response")
+
+    returnheaders=[]
+    _log("read_body_and_headers ---------------------------")
+    for header in info:
+        _log("read_body_and_headers "+header+"="+info[header])
+        returnheaders.append([header,info[header]])
+    handle.close()
+    _log("read_body_and_headers ---------------------------")
+
+    '''
+    # Lanza la petición
+    try:
+        response = urllib2.urlopen(req)
+    # Si falla la repite sustituyendo caracteres especiales
+    except:
+        req = urllib2.Request(url.replace(" ","%20"))
+    
+        # Añade las cabeceras
+        for header in headers:
+            req.add_header(header[0],header[1])
+
+        response = urllib2.urlopen(req)
+    '''
+    
+    # Tiempo transcurrido
+    fin = time.clock()
+    _log("read_body_and_headers Downloaded in %d seconds " % (fin-inicio+1))
+    _log("read_body_and_headers body="+data)
+
+    return data,returnheaders
+
+class NoRedirectHandler(urllib2.HTTPRedirectHandler):
+    def http_error_302(self, req, fp, code, msg, headers):
+        infourl = urllib.addinfourl(fp, headers, req.get_full_url())
+        infourl.status = code
+        infourl.code = code
+        return infourl
+    http_error_300 = http_error_302
+    http_error_301 = http_error_302
+    http_error_303 = http_error_302
+    http_error_307 = http_error_302
+
+# Parse string and extracts multiple matches using regular expressions
+def find_multiple_matches(text,pattern):
+    _log("find_multiple_matches pattern="+pattern)
+    
+    matches = re.findall(pattern,text,re.DOTALL)
+
+    return matches
+
+# Parse string and extracts first match as a string
+def find_single_match(text,pattern):
+    _log("find_single_match pattern="+pattern)
+
+    result = ""
+    try:    
+        matches = re.findall(pattern,text, flags=re.DOTALL)
+        result = matches[0]
+    except:
+        result = ""
+
+    return result
+
+def add_item( action="" , title="" , plot="" , url="" , thumbnail="" , fanart="" , show="" , episode="" , extra="", page="", info_labels = None, isPlayable = False , folder=True ):
+    _log("add_item action=["+action+"] title=["+title+"] url=["+url+"] thumbnail=["+thumbnail+"] fanart=["+fanart+"] show=["+show+"] episode=["+episode+"] extra=["+extra+"] page=["+page+"] isPlayable=["+str(isPlayable)+"] folder=["+str(folder)+"]")
+
+    listitem = xbmcgui.ListItem( title, iconImage="DefaultVideo.png", thumbnailImage=thumbnail )
```

Comment:
`iconImage` and `thumbnailImage` are both deprecated. Please use `setArt` instead. Solution below:

```python
listitem = xbmcgui.ListItem( title )
listitem.setArt({"thumb":"DefaultVideo.png", "icon":"DefaultVideo.png"})
```